lib/matplotlib/ticker.py

```diff
@@ -2583,6 +2585,113 @@ def view_limits(self, vmin, vmax):
         return result
 
 
+class AsinhLocator(Locator):
+    """
+    An axis tick locator specialized for the inverse-sinh scale
+
+    This is very unlikely to have any use beyond
+    the `~.scale.AsinhScale` class.
+    """
+    def __init__(self, linear_width, numticks=11, symthresh=0.2,
+                 base=0, subs=None):
+        """
+        Parameters
+        ----------
+        linear_width : float
+            The scale parameter defining the extent
+            of the quasi-linear region.
+        numticks : int, default: 11
+            The approximate number of major ticks that will fit
+            along the entire axis
+        symthresh : float, default: 0.2
+            The fractional threshold beneath which data which covers
+            a range that is approximately symmetric about zero
+            will have ticks that are exactly symmetric.
+        base : int, default: 0
```

Comment:
I find the base 0 a bit odd. I'd put default: 10 and then special case the 10 below?